wave_matching/src/icp.cpp

```diff
@@ -75,4 +79,367 @@ bool ICPMatcher::match() {
     return false;
 }
 
+void ICPMatcher::estimate_info() {
+    switch (this->estimate_method) {
+        case ICPMatcher::covar_method::LUM: this->estimate_lum();
+        case ICPMatcher::covar_method::CENSI: this->estimate_censi();
+        default: return;
+    }
+}
+
+// Taken from the Lu and Milios matcher in PCL
+void ICPMatcher::estimate_lum() {
+    if (this->icp.hasConverged()) {
+        auto list = this->icp.correspondences_.get();
+        Mat6 MM = Mat6::Zero ();
+        Vec6 MZ = Vec6::Zero ();
+        std::vector <Eigen::Vector3f> corrs_aver;
+        std::vector <Eigen::Vector3f> corrs_diff;
```

Comment:
This bit was taken from PCL and they used floats. Don't think it really matters one way or the other, the extra precision of double isn't going to change anything.